lib/database_cleaner/active_record/base.rb

```diff
@@ -15,9 +15,9 @@ def self.config_file_location
 
     class Base < DatabaseCleaner::Strategy
       def self.migration_table_name
-        if Gem::Version.new("7.1.0") < ::ActiveRecord.version
+        if ::ActiveRecord::Base.connection_pool.respond_to?(:schema_migration) # Rails >= 7.2
           ::ActiveRecord::Base.connection_pool.schema_migration.table_name
-        elsif Gem::Version.new("6.0.0") <= ::ActiveRecord.version
+        elsif ::ActiveRecord::Base.connection.respond_to?(:schema_migration) # Rails >= 6.0
```

Comment:
opinionated change in a separate commit, feel free to remove the last commit to use a version-based comparison